pkg/admission/clusterworkspacetypeexists/admission.go

```diff
@@ -256,7 +268,13 @@ func (o *clusterWorkspaceTypeExists) resolveType(ref tenancyv1alpha1.ClusterWork
 	if apierrors.IsNotFound(err) {
 		return nil, fmt.Errorf("spec.type %s does not exist", ref.String())
 	}
-	return cwt, err
+	if err != nil {
+		return nil, err
+	}
+	if !conditions.IsTrue(cwt, tenancyv1alpha1.ClusterWorkspaceTypeExtensionsResolved) {
```

Comment:
(no need for change) IMO this pattern is not good. Conditions should never be part of logic otherside of conditions themselves. This should really be reflected in some phase string.